include/deal.II/base/derivative_form.h

```diff
@@ -52,7 +53,7 @@ DEAL_II_NAMESPACE_OPEN
  *
  * @author Sebastian Pauletti, 2011, Luca Heltai, 2015
  */
-template <int order, int dim, int spacedim, typename Number=double>
+template <int order, int dim, int spacedim, typename Number>
```

Comment:
I don't have time this evening to look through the whole rest, but have you checked what doxygen makes of this? Does it still show the default argument?